File: mc/rmcp/src/main/java/calebxzhou/rdi/mc/common2/mcp/RMcpInventorySwapData.java
package calebxzhou.rdi.mc.common2.mcp;

public record RMcpInventorySwapData(
        String format,
        String from,
        String to,
        boolean dryRun,
        boolean changed,
        RMcpInventoryData.Item beforeFrom,
        RMcpInventoryData.Item beforeTo,
        RMcpInventoryData.Item afterFrom,
        RMcpInventoryData.Item afterTo,
        RMcpInventoryData inventory
) {
}

File: mc/rmcp/common/src/main/kotlin/calebxzhou/rdi/mc/common2/mcp/RMcpInventorySwapData.kt
package calebxzhou.rdi.mc.common2.mcp

@JvmRecord
data class RMcpInventorySwapData(
    val from: String,
    val to: String,
    val dryRun: Boolean,
    val changed: Boolean,
    val beforeFrom: RMcpInventoryData.Item,
    val beforeTo: RMcpInventoryData.Item,
    val afterFrom: RMcpInventoryData.Item,
    val afterTo: RMcpInventoryData.Item,
    val inventory: RMcpInventoryData?
)
